src/systemd/wait_online.rs

```diff
@@ -0,0 +1,258 @@
+use anyhow::{bail, ensure, Context, Result};
+use freedesktop_entry_parser::parse_entry;
+use regex::RegexBuilder;
+use std::time::Duration;
+
+macro_rules! service_file_path {
+    () => {{
+        const SERVICE_FILE_PATH_DEFAULT: &'static str =
+            "/lib/systemd/system/systemd-networkd-wait-online.service";
+        std::env::var("WAIT_ONLINE_SERVICE_FILE_PATH")
+            .unwrap_or(SERVICE_FILE_PATH_DEFAULT.to_string())
+    }};
+}
+
+macro_rules! env_file_path {
+    () => {{
+        const ENV_FILE_PATH_DEFAULT: &'static str = "/etc/omnect/systemd-networkd-wait-online.env";
+        std::env::var("ENV_FILE_PATH").unwrap_or(ENV_FILE_PATH_DEFAULT.to_string())
+    }};
+}
```

Comment:
Why as separate macros and not via lazy_static?

Reply:
the reason is testability: in some testcases we change the file paths. this doesn't work with lazy_static. I couldn't find another solution yet.